src/io/network/addrinfo.cpp

```diff
@@ -9,34 +9,50 @@
 // by the Apache License, Version 2.0, included in the file
 // licenses/APL.txt.
 
-#include <netdb.h>
-#include <cstring>
-
 #include "io/network/addrinfo.hpp"
 
 #include "io/network/network_error.hpp"
+#include "utils/iterator_concepts.hpp"
 
 namespace memgraph::io::network {
 
-AddrInfo::AddrInfo(struct addrinfo *info) : info(info) {}
+static_assert(utils::LegacyForwardIterator<AddrInfo::Iterator>);
```

Comment:
Let's slightly change this so we can get rid of `iterator_concepts.hpp` all alone. 
1. delete `iterator_concepts.hpp` 
2. Use this assert (see explanation below)
```
static_assert(std::input_iterator<AddrInfo::Iterator> && std::equality_comparable<AddrInfo::Iterator>);
```

`std::input_iterator` is a slightly different concept, i.e., it has slightly different requirements than `LegacyInputIterator`. The most important is that `ranges` work with sentinels to denote the end of the range and these are not part of the iterator interface but rather added as extra requirements to the ranges algorithms. Hence, `std::equality_comparable` is required in the assertion to mimic the requirements of `LegacyInputIterator`.

Note, that `std::input_iterator` has also stronger type requirements that `LegacyInputIterator` (see cpp ref) but are satisfied and not an insue in our case :)